src/main/java/com/lootfood/repository/lootpoint/LootPointRepositoryCustomImpl.java

```diff
@@ -0,0 +1,50 @@
+package com.lootfood.repository.lootpoint;
+
+import com.lootfood.entity.LootPoint;
+import com.mongodb.client.FindIterable;
+import com.mongodb.client.model.Filters;
+import lombok.RequiredArgsConstructor;
+import org.bson.Document;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageImpl;
+import org.springframework.data.domain.Pageable;
+import org.springframework.data.mongodb.core.MongoTemplate;
+import org.springframework.data.mongodb.core.query.Criteria;
+import org.springframework.data.mongodb.core.query.Query;
+import org.springframework.data.mongodb.core.query.Update;
+
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+
+@RequiredArgsConstructor
+public class LootPointRepositoryCustomImpl implements LootPointRepositoryCustom {
+    private final MongoTemplate mongoTemplate;
+
+    @Override
+    public Page<LootPoint> findAllLootPointsInPolygon(List<List<Double>> points, Pageable pageable) {
```

Comment:
It is wrong realization, you should rewrite it:
- Use pagination on the query to filter, order and fetch exactly that records that we are wanted
- not clear why you should call `findById` if you have selected LootPoint object